### PR TITLE
[receiver/splunkhecreceiver] [exporter/splunkhecexporter] add tls tests to hec receiver and exporter

### PR DESCRIPTION
**Pull request by timannguyen**
**Description:** <Describe what has changed.>

Adding additional tests to verify the following cases tls:
- Happy path
- Unhappy path:
  - Client cert is not signed by CA
  - Client cert is expired 
  - Client cert is using insecure SHA-1 algorithm

**Link to tracking Issue:** [21104](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21104)

**Testing:** <Describe what testing was performed and which tests were added.>

tls tests:
- Happy path
- Unhappy path
   - Client cert is not signed by CA
   - Client cert is expired 
   - Client cert is using insecure SHA-1 algorithm

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by atoulme** [comment]
Marking as Skip Changelog since this is only adding tests.

File: .chloggen/add tls tests to hec receiver and exporter.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: splunkhecreceiver, splunkhecexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: add tls tests to hec receiver and exporter
```
Comment (atoulme) [inline .chloggen/add tls tests to hec receiver and exporter.yaml:8]: no need for a changelog if no functional changes are present.

Comment (timannguyen) [reply .chloggen/add tls tests to hec receiver and exporter.yaml]: thank you. Just notice the tag.

Comment (atoulme) [reply .chloggen/add tls tests to hec receiver and exporter.yaml]: No problem. You can also have your PR title start with `[chore]` to skip changelog.

**Comment by dmitryax** [comment]
Do we cover anything on top of what's covered in https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confighttp ? If not, we should extend those tests instead

**Comment by dmitryax** [comment]
Splunk HEC is not the only exporter using `confighttp.HTTPClientSettings`, so the tests for that config should not be added here. I don't think there is anything else specific to Splunk HEC exporter in this test

**Comment by timannguyen** [comment]
> Do we cover anything on top of what's covered in https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confighttp ? If we do, we should extend those tests instead

@dmitryax hmm, there isn't anything new tested. I didn't realize this is tested there in http library instead. i'll close this PR out. thank you for looking through.

**Pull request closed without merging**